### DISCUSSION TITLE
Support promotion builds

### DISCUSSION THREAD

**Discussion by everflux** [woodpecker-ci/woodpecker#3592] [Request a Feature] [answered]
In Drone CI and other build systems it is possible to 'promote' a build.
This is either triggered by the UI or API, leading to executing the workflow(s) again for the commit id with the 'promote' event set, enabling conditional steps.
It could be used to promote a build from staging to production after manual acceptance tests / quality gates.

**Comment by anbraten** [comment]
I think this is what depoly is used for in woodpecker.

**Reply by everflux** [reply]
That sounds good, I found the event for it, but how can I trigger it from the ui?

**Reply by qwerty287** [reply]
If you have a successful pipeline, there's a "deploy" button next to the restart button.

**Reply by everflux** [reply]
Should that be in place in 2.4.1 already?

**Reply by qwerty287** [reply, marked as answer]
Yes, but currently only for push, tag and release pipelines. In 2.5.0 you'll be able to use any pipeline but it must be enabled in settings first.

**Reply by everflux** [reply]
I was missing it from manual execution which I used for testing - thanks a lot for the clarification!
